### PR TITLE
[WIp] Action: connectWallet added

### PR DESCRIPTION
**Pull request by snehmehta**
Ticket: https://github.com/EnsembleUI/ensemble/issues/208
Ability to connect with crypto wallet.

Sample YAML configuration:
```yaml
- Button:
    label: Connect to wallet
    onTap:
      action: connectWallet
      id: wallet
      wcProjectId: 'abc' # create from [here](https://cloud.walletconnect.com/sign-in). 
      metadata:
        name: Ensemeble app
        description: Ensemble forever
        url: https://ensembleui.com/
        iconUrl: https://ensembleui.com/assets/images/logo.svg
      onComplete: |
        //@code
        ensemble.storage.publicKey = wallet.publicKey;
      onError: |
        //@code
        ensemble.debug('something went wrong')

```
TODO:

- [ ] Check and handle IOS deeplink.
- [ ] Adding action in ensemble_schema.json.
- [ ] Testing

### CODE REVIEW

File: lib/framework/action.dart [hunk 1]
```diff
@@ -201,7 +201,44 @@ class FileUploadAction extends EnsembleAction {
...
+}
+
+class AppMetaData {
+  String name;
+  String description;
+  String url;
+  String iconUrl;
+
+  AppMetaData({
+    required this.name,
```
Comment (vusters) [inline lib/framework/action.dart:228]: What kind of errors do you get if these required fields are not specified? If the errors are not shown in the Editor, try throwing an LanguageError

File: lib/framework/data_context.dart [hunk 2]
```diff
@@ -694,3 +696,58 @@ class File {
...
+
+  Map<String, SessionData>? allSessions() {
+    return _wcClient.getActiveSessions();
+  }
+
+  signTransaction() async {
+    final _ = await _wcClient.request(
+      topic: _sessionData!.topic, 
+      chainId: ChainUtils.ethereumNamespace, 
+      request: SessionRequestParams(method: ChainUtils.ethSignTransaction, params: 'abc')
```
Comment (vusters) [inline lib/framework/data_context.dart:750]: abc = input?

**Comment by kmahmood74** [comment]
@snehmehta once you have addressed the comments Vu raised and once the PR is approved, please merge it into main. It's ok to have this in main even if it is incomplete, we just won't expose it yet.

**Comment by snehmehta** [comment]
#392

**Pull request closed without merging**